pkg/capi/aws.go

```diff
@@ -143,6 +143,81 @@ var _ = Describe("Cluster API AWS MachineSet", framework.LabelCAPI, framework.La
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-78677", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+	})
+
+	//huliu-OCP-75662 - [CAPI] AWS Machine API Support of more than one block device.
+	It("should be able to run a machine with more than one block device", func() {
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		encryptBool := true
+		unencryptBool := false
+		awsMachineTemplate.Spec.Template.Spec.NonRootVolumes = []awsv1.Volume{
+			{
+				DeviceName: "/dev/xvda",
+				Size:       120,
+				Type:       awsv1.VolumeTypeGP3,
+				IOPS:       5000,
+				Encrypted:  &encryptBool,
+			},
+			{
+				DeviceName: "/dev/sdf",
+				Size:       120,
+				Type:       awsv1.VolumeTypeGP2,
+				Encrypted:  &unencryptBool,
+			},
+		}
+		Expect(cl.Create(ctx, awsMachineTemplate)).To(Succeed(), "Failed to create awsmachinetemplate")
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-75662", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+	})
+
+	//huliu-OCP-75663 - [CAPI] User defined tags can be applied to AWS EC2 Instances.
+	It("should be able to run a machine with user defined tags", func() {
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		awsMachineTemplate.Spec.Template.Spec.AdditionalTags = map[string]string{
+			"adminContact": "qe",
+			"costCenter":   "1981",
+			"customTag":    "test",
+			"Email":        "qe@redhat.com",
+		}
+		Expect(cl.Create(ctx, awsMachineTemplate)).To(Succeed(), "Failed to create awsmachinetemplate")
+		machineSetParams = framework.UpdateCAPIMachineSetName("aws-machineset-75663", machineSetParams)
+		machineSet, err = framework.CreateCAPIMachineSet(ctx, cl, machineSetParams)
+		Expect(err).ToNot(HaveOccurred(), "Failed to create CAPI machineset")
+		framework.WaitForCAPIMachinesRunning(ctx, cl, machineSet.Name)
+	})
+
+	//OCP-76794 - [CAPI] Support AWS capacity-reservations in CAPA.
+	It("should be able to run a machine with capacity-reservations", framework.LabelQEOnly, func() {
+		awsMachineTemplate = newAWSMachineTemplate(mapiDefaultProviderSpec)
+		By("Access AWS to create CapacityReservation")
+		awsClient := framework.NewAwsClient(framework.GetCredentialsFromCluster(oc))
+		capacityReservationID, err := awsClient.CreateCapacityReservation(mapiDefaultProviderSpec.InstanceType, "Linux/UNIX", mapiDefaultProviderSpec.Placement.AvailabilityZone, 1)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(capacityReservationID).ToNot(Equal(""))
+
+		defer func() {
```

Comment:
I wonder if we should try to use `DeferCleanup` from the ginkgo package for this?

Reply:
Sure, updated, thank you!